### ISSUE TITLE
String interpolation with function call not working for `meta` properties

### ISSUE THREAD

**Issue by TylerPachal** [hashicorp/nomad#9870]
### Nomad version
0.12.9

### Issue

String interpolation with function call not working for `meta` properties.

I have two places in my `.hcl` file where I am using the `substr` function to grab the short version of my commit hash, which is used to tag my docker image.  In the `meta` stanza, the string is not interpolated, and as show as-is in the UI (see the screenshot).  In the `task/config` stanza, it works fine.

### Job file 

```
job "my-job" {
  
  meta { 
    image_tag = "${substr("[[ .DEPLOY_SUB___VERSION ]]", 0, 7)}"           # This doesn't work, and is show in the UI as-is
  }

  group "my-group" {
    task "my-task" {
      config {
        image = "uri.for.my.image:${substr("[[ .DEPLOY_SUB___VERSION ]]", 0, 7)}"       # This works
      }
    }
  }

  # ...
} 
```

This is what is presented in the UI:
![nomad](https://user-images.githubusercontent.com/4924153/105387577-1359e500-5bec-11eb-8233-6dd0929fcbc5.png)

I also tried doing the following, but it was invalid `hcl`:
```
image_tag = substr("[[ .DEPLOY_SUB___VERSION ]]", 0, 7)
```

---

Edit: We are using Levant to do the replacement of `DEPLOY_SUB___VERSION`.

**Comment by tgross** [comment]
@TylerPachal the HCL2 functions like `substr` are only available in Nomad 1.0 and above, so Nomad is trying that as a plain old string. Even for Nomad 1.0 the syntax needs to be a little different because of a bug in quote-escaping (ref https://github.com/hashicorp/nomad/issues/9838), but this job works:


```hcl
job "example" {
  datacenters = ["dc1"]

  meta {
    tag = substr("SOMEVALUE", 0, 7)
  }

  group "cache" {
    network {
      port "db" {
        to = 6379
      }
    }

    task "redis" {
      driver = "docker"

      config {
        image = "redis:3.2"

        ports = ["db"]
      }

      resources {
        cpu    = 500
        memory = 256
      }
    }
  }
}
```

```
$ nomad job run ./example.nomad
==> Monitoring evaluation "2e7c51eb"
    Evaluation triggered by job "example"
    Allocation "5108edaf" created: node "5005e621", group "cache"
==> Monitoring evaluation "2e7c51eb"
    Evaluation within deployment: "88bcb501"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "2e7c51eb" finished with status "complete"

$ nomad alloc exec 5108 env | grep META
NOMAD_META_TAG=SOMEVAL
NOMAD_META_tag=SOMEVAL
```

**Comment by tgross** [comment]
>  Even for Nomad 1.0 the syntax needs to be a little different because of a bug in quote-escaping (ref https://github.com/hashicorp/nomad/issues/9838), but this job works:

Looks like #9838 was the wrong bug, as this works in master now. But looks like this is safe to close.
